### PR TITLE
Ports TG push broom

### PR DESCRIPTION
**Pull request by Henri215**
## What Does This PR Do

Port from https://github.com/tgstation/tgstation/pull/49120
Adds push broom for pushing trash into disposal bins!
Added 2 brooms to custodial closet, 1 for janitor ERT backpack, 1 for janitorial crate.
Push broom fits into janitor belt and janicart.
## Why It's Good For The Game

Push brooms are a great alternative if you dont wanna carry a big trashbag, a smaller item that demands more work.
## Images of changes

https://user-images.githubusercontent.com/77684085/216830828-0930e91a-84a2-42b8-8af3-0b547ee22bdd.mp4

## Testing

- See video above
## Changelog
:cl: Henri215, Mickyan
add: Added the push broom, used for sweeping trash into bins (fits into janibelt and janicart).
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -953,3 +953,70 @@
...
+				garbage.forceMove(target_bin)
+			else
+				garbage.Move(new_item_loc, user.dir)
+			trash_amount++
+		if(trash_amount > BROOM_PUSH_LIMIT)
+			break
+	if(trash_amount > 1)
+		if(target_bin)
+			target_bin.update_icon()
+			to_chat(user, "<span class='notice'>You sweep the pile of garbage into [target_bin].")
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/twohanded.dm:1015]:
```suggestion
			to_chat(user, "<span class='notice'>You sweep the pile of garbage into [target_bin].</span>")
```

**Review by S34NW — CHANGES_REQUESTED** [review]
Your in-hand icon states shown in the video look like the reverse of what I would expect

**Comment by Henri215** [comment]
> Your in-hand icon states shown in the video look like the reverse of what I would expect

This video was recorded before i fixed it.

File: code/game/objects/items/weapons/twohanded.dm [hunk 2]
```diff
@@ -953,3 +953,70 @@
...
+	force_wielded = 12
+	attack_verb = list("swept", "brushed off", "bludgeoned", "whacked")
+	resistance_flags = FLAMMABLE
+
+/obj/item/twohanded/push_broom/update_icon_state()
+	icon_state = "broom[wielded]"
+
+/obj/item/twohanded/push_broom/wield(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>You brace the [src] against the ground in a firm sweeping stance.</span>")
```
Comment (lewcc) [inline code/game/objects/items/weapons/twohanded.dm:982]:
```suggestion
	to_chat(user, "<span class='notice'>You brace [src] against the ground in a firm sweeping stance.</span>")
```

File: code/game/objects/structures/janicart.dm [hunk 3]
```diff
@@ -56,7 +57,12 @@
 				m.janicart_insert(user, src)
 			else
 				to_chat(user, fail_msg)
-
+		else if(istype(I, /obj/item/twohanded/push_broom))
+			if(!mybroom)
+				var/obj/item/twohanded/push_broom/b = I
+				b.janicart_insert(user, src)
```
Comment (lewcc) [inline code/game/objects/structures/janicart.dm:63]: Good convention to uppercase this if you're using a single letter
```suggestion
				var/obj/item/twohanded/push_broom/B = I
				B.janicart_insert(user, src)
```

File: code/game/objects/items/weapons/twohanded.dm [hunk 4]
```diff
@@ -953,3 +953,70 @@
...
+			break
+	if(trash_amount > 1)
+		if(target_bin)
+			target_bin.update_icon()
+			to_chat(user, "<span class='notice'>You sweep the pile of garbage into [target_bin].</span>")
+		playsound(loc, 'sound/weapons/thudswoosh.ogg', 10, TRUE, -1)
+
+/obj/item/twohanded/push_broom/proc/janicart_insert(mob/user, obj/structure/janitorialcart/J)
+	J.mybroom = src
+	J.put_in_cart(src, user)
```
Comment (lewcc) [inline code/game/objects/items/weapons/twohanded.dm:1020]: Can you use a variable name like `cart` instead of just `J`?

Comment (Henri215) [reply code/game/objects/items/weapons/twohanded.dm:1020]: Done

**Review by S34NW — APPROVED** [review]
On second thought I have given up on caring about the icon state issue, as it's not a critical interaction/action.

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/structures/janicart.dm [hunk 5]
```diff
@@ -12,8 +12,9 @@
 	//copypaste sorry
 	var/maximum_volume = 150
 	var/amount_per_transfer_from_this = 5 //shit I dunno, adding this so syringes stop runtime erroring. --NeoFite
-	var/obj/item/storage/bag/trash/mybag	= null
+	var/obj/item/storage/bag/trash/mybag = null
 	var/obj/item/mop/mymop = null
+	var/obj/item/twohanded/push_broom/mybroom = null
```
Comment (SteelSlayer) [inline code/game/objects/structures/janicart.dm:17]: Need a `QDEL_NULL(mybroom)` in destroy.

**Review by SteelSlayer — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/janicart.dm (hunk 5 above)
Comment (Henri215) [reply code/game/objects/structures/janicart.dm:17]: Done

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**